libass/ass_cache.c

```diff
@@ -99,7 +99,7 @@ const CacheDesc bitmap_cache_desc = {
 
 
 // composite cache
-static uint32_t composite_hash(void *key, uint32_t hval)
+static ass_hashcode composite_hash(void *key, ass_hashcode hval)
```

Comment:
I think this is perfectly described by "give the hash a more generic name" and should go to the first commit.
In that case the last commit would be naturally localized in `ass_utils.h` as it should be.